READMEs/publish-flow-onchain.md

```diff
@@ -50,4 +68,42 @@ That's it! You've created a data asset of "SmartContractCall" asset type. It inc
 
 ## 3.  Consume the dataset
 
-(Consume here is just like in [consume-flow](READMEs/consume-flow.md]. The file downloaded is a .json. From that, use the python `json` library to parse it as desired.)
+```python
+# Alice mints and send datatokens to Bob
+to_address = bob_wallet.address
+amt_tokens = ocean.to_wei(10)  # just need 1, send more for spare
+datatoken.mint(to_address, amt_tokens, alice_wallet)
+
+
+# userdata: parameters required for the contract call 
+WETH = "0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6"  # GOERLI WETH
+UNI = "0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984"  # UNI GOERLI
+userdata = {
+    "amountIn": ocean.to_wei(1),
+    "path": [WETH, UNI]
+}
+
+# pay for acces to the service
+order_tx_id = ocean.assets.pay_for_access_service(
+    asset,
+    bob_wallet,
+    service,
+    consume_market_order_fee_address=bob_wallet.address,
+    consume_market_order_fee_token=datatoken.address,
+    consume_market_order_fee_amount=0,
+    userdata=userdata
+)
```

Comment:
In making this call, it gives a traceback:
```text
NameError: name 'service' is not defined
```

Reply:
Done